File: my-code/509/main.go
package main

import "fmt"

// @Time     : 2021/7/22 21:35
// @Author   : Ranshi
// @File     : main.go

func fib(n int) int {
	if n == 0 {
		return 0
	} else if n < 2 {
		return 1
	}
	a, b := 1, 0
	for i := 2; i <= n; i++ {
		a, b = a+b, a
	}
	return a
}

func main() {
	fmt.Println(fib(4))
}

File: my-code/509/main.py
# -*- coding:utf-8 -*-
# @Time     : 2021/05/28 18: 10
# @Author   : Ranshi
# @File     : 509. 斐波那契数列.py
import math


class Solution:
    def fib(self, N: int) -> int:
        return round(pow((1 + math.sqrt(5)) / 2, N) / math.sqrt(5))


if __name__ == "__main__":
    s = Solution()
    n = input()
    print(s.fib(int(n)))
